src/main/scala/org/constellation/API.scala

```diff
@@ -98,6 +103,64 @@ class API(udpAddress: InetSocketAddress)(implicit system: ActorSystem, val timeo
   val getEndpoints: Route =
     extractClientIP { clientIP =>
       get {
+        path("channels") {
+          complete(dao.threadSafeMessageMemPool.activeChannels.keys.toSeq)
+        } ~
+        path ("channel" / Segment) { channelHash =>
+
+
+          val maxDepth = 10
+
+          def getMessageWithSnapshotHash(
```

Comment:
Would be nice to extract this to a method on the class

Reply:
As a general note - although I don't know what's common in the network field with Scala - one may reserve the "get" prefix for actual getters of members. 
If there can be non-trivial computational overhead/latency associated with a classes method call, one may term them "compute". E.g. when a method were to compute a hash of a dynamically sized list member of an object `obj`, then you're `obj.computeHash()` instead of `obj.getHash()`.